### PR TITLE
Add support for challenge cookie domain field

### PR DESCRIPTION
**Pull request by arbelaxRbz**
Read custom.json file as well, parse ‘sites’ and save in the config object.
Add and pass server group id to RInfo (like secpolid).
Add “Domain=<val>” to the rbzid cookie, depending on the new field’s content.

### CODE REVIEW

**Review by olgasobolevsky — DISMISSED** [review]

File: curiefense/curieproxy/rust/curiefense-lua/src/lib.rs [hunk 1]
```diff
@@ -73,6 +75,9 @@ fn lua_convert_args<'l>(lua: &'l Lua, args: LuaTable<'l>) -> Result<LuaArgs<'l>,
     let vsecpolid = args
         .get("secpolid")
         .map_err(|_| "Missing log level argument".to_string())?;
+    let vsergrpid = args
+        .get("sergrpid")
+        .map_err(|_| "Missing log level argument".to_string())?;
```
Comment (Freakachoo) [inline curiefense/curieproxy/rust/curiefense-lua/src/lib.rs:80]: log level argument? Or missing `sergrpid`? Not sure

File: curiefense/curieproxy/rust/curiefense-lua/src/lib.rs [hunk 2]
```diff
@@ -108,6 +113,10 @@ fn lua_convert_args<'l>(lua: &'l Lua, args: LuaTable<'l>) -> Result<LuaArgs<'l>,
         Err(rr) => return Err(format!("Could not convert the hops argument: {}", rr)),
         Ok(i) => i,
     };
+    let sergrpid = match FromLua::from_lua(vsergrpid, lua) {
+        Err(rr) => return Err(format!("Could not convert the hops argument: {}", rr)),
```
Comment (Freakachoo) [inline curiefense/curieproxy/rust/curiefense-lua/src/lib.rs:117]: "hops" is from line 108. I think here is "sergrpid" (same for secpolid)

File: curiefense/curieproxy/rust/curiefense/src/config/custom.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+use crate::config::raw::RawSite;
+use crate::logs::Logs;
+
+/// Contains objects for the custom.json file
+
+#[derive(Debug, Clone, serde::Deserialize)]
+pub struct Site {
+    pub id: String,
+    pub name: String,
+    // pub mobile_sdk: String,
```
Comment (Freakachoo) [inline curiefense/curieproxy/rust/curiefense/src/config/custom.rs:12]: not needed?

File: curiefense/curieproxy/rust/curiefense/src/config/custom.rs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+impl Site {
+    pub fn resolve(logs: &mut Logs, raw_sites: Vec<RawSite>) -> HashMap<String, Site> {
+        let mut sites_map: HashMap<String, Site> = HashMap::new();
+        for raw_site in raw_sites {
+            let challenge_cookie_domain = raw_site
+                .challenge_cookie_domain
+                .as_ref()
+                .map(|domain| {
+                    if domain.is_empty() {
+                        "$host".to_string()
```
Comment (Freakachoo) [inline curiefense/curieproxy/rust/curiefense/src/config/custom.rs:36]: maybe move "$host" into a constant?

File: curiefense/curieproxy/rust/curiefense/src/config/mod.rs [hunk 5]
```diff
@@ -497,6 +563,7 @@ impl Config {
         let rawcontentfilterprofiles = Config::load_config_file(&mut logs, &bjson, "contentfilter-profiles.json");
         let flows = Config::load_config_file(&mut logs, &bjson, "flow-control.json");
         let virtualtags = Config::load_config_file(&mut logs, &bjson, "virtual-tags.json");
+        let rawsites: Vec<RawSite> = Config::load_custom_config_file(&mut logs, &bjson, "custom.json");
```
Comment (Freakachoo) [inline curiefense/curieproxy/rust/curiefense/src/config/mod.rs:566]: we're getting "rawsites" but we're loading custom file. But I think in that file we have more stuff than just sites.
Could we maybe return some tuple from the function and for now just take "rawsites"?

File: curiefense/curieproxy/rust/curiefense/src/config/raw.rs [hunk 6]
```diff
@@ -88,6 +88,24 @@ pub struct RawSecurityPolicy {
     pub limit_ids: Vec<String>,
 }
 
+/// a mapping of the custom document
```
Comment (Freakachoo) [inline curiefense/curieproxy/rust/curiefense/src/config/raw.rs:91]: I think this is the mapping only for sites, not for the whole custom doc

**Review by Freakachoo — APPROVED** [review]

**Review by olgasobolevsky — APPROVED** [review]

**Pull request merged**